app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/adapter/EditItemAdapterTest.java

```diff
@@ -1,96 +1,120 @@
-import org.robolectric.RuntimeEnvironment;
 import org.robolectric.annotation.Config;
 
 import java.util.ArrayList;
 import java.util.List;
 
-import static com.invisibleteam.goinvisible.util.TagsMatcher.containsItem;
 import static com.invisibleteam.goinvisible.util.TagsMatcher.containsKey;
+import static com.invisibleteam.goinvisible.util.TagsMatcher.containsSectionName;
 import static com.invisibleteam.goinvisible.util.TagsMatcher.containsTag;
 import static com.invisibleteam.goinvisible.util.TagsMatcher.containsValue;
 import static com.invisibleteam.goinvisible.util.TagsMatcher.notContainsTag;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.is;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 @RunWith(RobolectricTestRunner.class)
 @Config(constants = BuildConfig.class, sdk = 21)
 public class EditItemAdapterTest {
 
-    private Tag TAG1 = createTag("key1", "value1");
-    private Tag TAG2 = createTag("key2", "value2");
-    private Tag TAG3 = createTag("key3", "value3");
-    private Tag TAG4 = createTag("key4", "value4");
-    private Tag TAG5 = createTag("key5", "value5");
-    private Tag TAG6 = createTag("key6", "value6");
-    private Tag TAG7 = createTag("key7", "value7");
+    private static final int DEFAULT_VIEW_TYPE = -1;
+    private static final int NUMBER_OF_HEADERS = 4;
+    private final Tag tag1 = createTag("key1", "value1", TagGroupType.IMAGE_INFO);
+    private final Tag tag2 = createTag("key2", "value2", TagGroupType.LOCATION_INFO);
+    private final Tag tag3 = createTag("key3", "value3", TagGroupType.DEVICE_INFO);
+    private final Tag tag4 = createTag("key4", "value4", TagGroupType.ADVANCED);
 
     private final List<Tag> tagList = new ArrayList<Tag>() {
         {
-            add(TAG1);
-            add(TAG2);
-            add(TAG3);
-            add(TAG4);
-            add(TAG5);
-            add(TAG6);
-            add(TAG7);
+            add(tag1);
+            add(tag2);
+            add(tag3);
+            add(tag4);
         }
     };
-    private Context context;
+    private EditActivity activity;
 
     @Before
     public void init() {
-        context = RuntimeEnvironment.application;
+        activity = Robolectric.buildActivity(EditActivity.class).create().get();
     }
 
     @Test
-    public void whenTagsListIsNotEmpty_ViewHolderIsBound() {
+    public void whenTagsListIsNotEmpty_TagViewHolderIsBound() {
         //Given
-        EditItemAdapter adapter = new EditItemAdapter();
+        EditItemAdapter adapter = new EditItemAdapter(new EditItemAdapterHelper());
         adapter.updateTagList(tagList);
-        ViewGroup view = mock(ViewGroup.class);
-        when(view.getContext()).thenReturn(context);
+        ViewGroup parent = new ViewGroup(activity) {
```

Comment:
Better is to use some implementation like for example LinearLayout. It helps you avoid unnecessary methods implementation.